src/equations/ideal_mhd_multiion_2d.jl

```diff
@@ -162,16 +162,18 @@ end
       f4 = rho_v1*v3 #- B1*B3
       f5 = (kin_en + gamma*p/(gamma - 1))*v1 + 2 * mag_en * vk1_plus[k] - B1 * (vk1_plus[k] * B1 + vk2_plus[k] * B2 + vk3_plus[k] * B3)
 
-      f = (f..., f1, f2, f3, f4, f5)
+      f[3 + (k - 1) * 5 + 1] = f1
+      f[3 + (k - 1) * 5 + 2] = f2
+      f[3 + (k - 1) * 5 + 3] = f3
+      f[3 + (k - 1) * 5 + 4] = f4
+      f[3 + (k - 1) * 5 + 5] = f5
```

Comment:
Could you please briefly comment on the memory layout here?

Reply:
I just translated what was there before, but I totally agree, this should be fixed before merging to `main`. I’ll leave it up to @amrueda to maybe use a `set_component` function or something similar that is more descriptive 🙂

Reply:
Added here: 57b8ea4